src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -308,5 +248,22 @@ internal static void ValidateCachePath(string path)
                 throw new ArgumentException($"The directory with path {directoryPath} does not exist.");
             }
         }
+
+        internal static void ValidateFileCacheExpiration(TimeSpan expiration)
+        {
+            if (expiration <= TimeSpan.Zero)
+            {
+                throw new ArgumentException($"Please provide a valid {nameof(OfflineFileCacheOptions)}.{nameof(OfflineFileCacheOptions.FileCacheExpiration)}.");

```

Comment:
Looks like it's also out of range.

Reply:
With ARgumentOutOfRange just need param name in constructor.

Reply:
This exception will be thrown even if someone doesn't set the FileCacheExpiration since we can't really check for null here. I could split the check and handle `expiration==TimeSpan.Zero` and `expiration<TimeSpan.Zero` separately, but even then, we can't pin point if user passed 0 expiration or just forgot to set it. So I just bundled those up into one `ArgumentException`